x-pack/plugins/security_solution/public/common/components/exceptions/helpers.tsx

```diff
@@ -487,17 +594,33 @@ export const entryHasNonEcsType = (
 export const defaultEndpointExceptionItems = (
   listId: string,
   ruleName: string,
-  alertEcsData: Ecs
+  alertEcsData: FlattenType<Ecs>
 ): ExceptionsBuilderExceptionItem[] => {
-  const { file, event: alertEvent } = alertEcsData;
+  const { file, process, Ransomware, event: alertEvent } = alertEcsData;
+  const eventCode = alertEvent && alertEvent.code ? alertEvent.code : '';
+
+  if (eventCode === 'ransomware') {
+    return getProcessCodeSignature(alertEcsData).map((codeSignature) =>
+      getPrepopulatedRansomwareException({
+        listId,
+        ruleName,
+        sha256Hash: process && process.hash && process.hash.sha256 ? process.hash.sha256 : '',
```

Comment:
I think it might be cleaner if `getPrepopulatedRansomwareException` took `alertEcsData` as an argument and dealt with retrieving the hash, executable, eventCode, and Ransomware information. Right now, `defaultEndpointExceptionItems` has some logic to pull some of these values out but so does `getPrepopulatedRansomwareException`.